frontend/frontend.go

```diff
@@ -151,8 +151,20 @@ func (f *Frontend) HealthzReady(writer http.ResponseWriter, request *http.Reques
 
 func (f *Frontend) ArmResourceListBySubscription(writer http.ResponseWriter, request *http.Request) {
 	ctx := request.Context()
-	logger := ctx.Value(ContextKeyLogger).(*slog.Logger)
-	versionedInterface := ctx.Value(ContextKeyVersion).(api.Version)
+	logger, ok := LoggerFromContext(ctx)
+	if !ok {
+		logger = DefaultLogger()
```

Comment:
do we really want to use the default logger though? If our context is not set properly, we should be aborting the request.

Reply:
I do agree with Spencer. This would be hiding issue. In this case I would prefer error and abort.

Reply:
Yes, it makes sense. I have changed that to now to error when the value is not in the context.

Reply:
In these frontend handler functions, we can just use the frontend's Logger instance directly (`f.logger`) and skip the error handling.  I don't know why I initially wrote these functions to fetch it from the context.  We stuff the logger into the context so middleware functions have access.

Reply:
@mbarnes: I just saw your comment. I am changing it now as it makes more sense.